packages/input-manager/src/inputManager.ts

```diff
@@ -109,14 +110,17 @@ class InputManager extends EventEmitter<DeviceEvents> {
 	}
 
 	private async createDevice(deviceId: string, deviceConfig: SomeDeviceConfig): Promise<void> {
-		let device
+		let createdDevice: Device | undefined = undefined
 		try {
 			this.#logger.debug(`Creating new device "${deviceId}"...`)
-			device = createNewDevice(deviceConfig, this.#logger.child({ deviceId }))
-			device.on('trigger', (eventArgs) => {
+			const device = createNewDevice(deviceConfig, this.#logger.child({ deviceId }))
+			createdDevice = device
+			device.on('trigger', () => {
+				// Device notifies us that a trigger has changed.
+
 				this.emit('trigger', {
-					...eventArgs,
 					deviceId,
+					getNextTrigger: () => device.getNextTrigger(),
```

Comment:
This setup is pretty weird, I think? Shouldn't the `inputManager` just have a method called `getNextTrigger` that then returns the next trigger, from across all devices? The API of emitting an event called `trigger` and then attaching a function, to that event object, called "getNextTrigger" is just very counterintuitive: get which trigger? This one? If not the one this event is describing, then why is it on the event object?